src/pamiq_core/model/interface.py

```diff
@@ -1,5 +1,8 @@
 from abc import ABC, abstractmethod
-from typing import Any
+from typing import Any, Generic, TypeVar
+
+# Define the type restricted to InferenceModel subclasses
+T = TypeVar("T", bound="InferenceModel")
```

Comment:
Python 3.12以降は 型変数が builtinの仕様になったのでいらないんだ
```suggestion
from typing import Any
```